python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -829,6 +889,29 @@ def _get_subprocess_kwargs(self):
 
         return kwargs
 
+    def _get_task_parent_entity_type(self, task_id):
+        """
+        Gets the Task entity's parent entity type.
+        """
+        cache = self.WSS_KEY_CACHE[self._wss_key]
+
+        if "task_parent_types" in cache and task_id in cache["task_parent_types"]:
+            logger.debug("Parent entity type found in cache for Task %s.", task_id)
+        else:
+            context = sgtk.context.from_entity(
+                self._engine.sgtk,
+                "Task",
+                task_id,
+            )
+            entity_type = context.entity["type"]
+
+            if "task_parent_types" in cache:
```

Comment:
Depending on whether you think the following is cleaner or not, you could also have written this as

```python
cache.setdefault("task_parent_types", {})[task_id] = entity_type
```